src/main/java/ActivityPresenter.java

```diff
@@ -30,13 +32,13 @@ public void onNext(Activity activity) {
         String nextActivity = activity.getNextActivity() + " can be done with "
                 + activity.getParticipants() + people;
         System.out.println(nextActivity);
-        view.setActivity(nextActivity);
-        view.setLink(activity.getLink());
+        viewProvider.get().setActivity(nextActivity);
+        viewProvider.get().setLink(activity.getLink());
```

Comment:
Call viewProvider.get() only once.

Reply:
What I mean is don't repeat it twice.
Call it once and save the returned value.